python/dgl/heterograph.py

```diff
@@ -4701,6 +4701,46 @@ def int(self):
                               self._node_frames,
                               self._edge_frames)
 
+    def reverse(self, share_ndata=True, share_edata=False):
```

Comment:
One possibility is to implement it directly using `dgl.heterograph` so that we can avoid accessing private properties of the graph.